batcher/aligned-batcher/src/types/batch_state.rs

```diff
@@ -117,42 +97,115 @@ impl BatchState {
         None
     }
 
-    /// Updates the user with address `addr` with the provided values of `new_nonce`, `new_min_fee` and
-    /// `new_proof_count`.
+    pub(crate) fn update_user_nonce(&mut self, addr: &Address, new_nonce: U256) -> Option<U256> {
+        if let Entry::Occupied(mut user_state) = self.user_states.entry(*addr) {
+            user_state.get_mut().nonce = new_nonce;
+            return Some(new_nonce);
+        }
+        None
+    }
+
+    pub(crate) fn update_user_total_fees_in_queue(
+        &mut self,
+        addr: &Address,
+        new_total_fees_in_queue: U256,
+    ) -> Option<U256> {
+        if let Entry::Occupied(mut user_state) = self.user_states.entry(*addr) {
+            user_state.get_mut().total_fees_in_queue = new_total_fees_in_queue;
+            return Some(new_total_fees_in_queue);
+        }
+        None
+    }
+
+    pub(crate) fn update_user_total_fees_in_queue_of_replacement_message(
+        &mut self,
+        addr: &Address,
+        original_max_fee: U256,
+        new_max_fee: U256,
+    ) -> Option<U256> {
+        let fee_difference = new_max_fee - original_max_fee; //here we already know new_max_fee > original_max_fee
+        if let Entry::Occupied(mut user_state) = self.user_states.entry(*addr) {
+            user_state.get_mut().total_fees_in_queue += fee_difference;
+            return Some(user_state.get().total_fees_in_queue);
+        }
+        None
+    }
+
+    /// Updates the user with address `addr` with the provided values of
+    /// `new_nonce`, `new_max_fee_limit`, `new_proof_count` and `new_total_fees_in_queue`
     /// If state is updated successfully, returns the updated values inside a `Some()`
     /// If the address was not found in the user states, returns `None`
     pub(crate) fn update_user_state(
         &mut self,
         addr: &Address,
         new_nonce: U256,
-        new_min_fee: U256,
+        new_max_fee_limit: U256,
         new_proof_count: usize,
-    ) -> Option<(U256, U256, usize)> {
+        new_total_fees_in_queue: U256,
+    ) -> Option<(U256, U256, usize, U256)> {
```

Comment:
Returning an Option with the same values it receives seems redundant and odd. It would be better to return a Result here. If it is too complicated to refactor let's leave it for a future pr.

Reply:
I agree. I impemented it this way to mimick the other `update_x()` funtions.
I would leave this refactor to another PR, and refactor all of them.
I have added TODO comment